lib/ssz_ex/decode.ex

```diff
@@ -56,115 +55,137 @@ defmodule SszEx.Decode do
     {:ok, element}
   end
 
-  defp decode_uint(_binary, size), do: {:error, "invalid byte size #{inspect(size)}"}
+  defp decode_uint(binary, size),
+    do:
+      {:error,
+       "Invalid binary length while decoding uint.\nExpected size: #{size}.\nFound:#{bit_size(binary)}\nBinary:#{inspect(binary)}"}
 
   defp decode_bool("\x01"), do: {:ok, true}
   defp decode_bool("\x00"), do: {:ok, false}
-  defp decode_bool(_), do: {:error, "invalid bool value"}
 
-  defp decode_bitlist(bit_list, max_size) when bit_size(bit_list) > 0 do
+  defp decode_bool(binary),
+    do:
+      {:error,
+       "Invalid binary value while decoding bool.\nExpected value: x01/x00.\nFound: x#{Base.encode16(binary)}."}
```

Comment:
We should split this error in two:
1. the binary is `<<x>>` with `x` in 2..255 (in this case print `x`)
2. the binary doesn't have a length of 1 (in this case we print the length)